src/services/oauth2/index.js

```diff
@@ -0,0 +1,186 @@
+import Debug from 'debug';
+import errors from 'feathers-errors';
+import passport from 'passport';
+import { exposeConnectMiddleware } from '../../middleware';
+import { successfulLogin } from '../../middleware';
+
+const debug = Debug('feathers-authentication:oauth2');
+const defaults = {
+  successRedirect: '/auth/success',
+  passwordField: 'password',
+  userEndpoint: '/users',
+  tokenEndpoint: '/auth/token',
+  passReqToCallback: true,
+  callbackSuffix: 'callback',
+  permissions: {}
+};
+
+export class Service {
+  constructor(options = {}) {
+    this.options = options;
+  }
+
+  oauthCallback(req, accessToken, refreshToken, profile, done) {
+    let app = this.app;
+    const options = this.options;
+    const params = {
+      internal: true,
+      query: {
+        // facebookId: profile.id
+        [`${options.provider}Id`]: profile.id
+      }
+    };
+
+    // console.log('Authenticating', accessToken, refreshToken, profile);
+
+    // Find or create the user since they could have signed up via facebook.
+    app.service(options.userEndpoint)
+      .find(params)
+      .then(users => {
+        // Paginated services return the array of results in the data attribute.
+        let user = users[0] || users.data && users.data[0];
+
+        // If user found return them
+        if (user) {
+          return done(null, user);
+        }
+
+        // No user found so we need to create one.
+        // 
+        // TODO (EK): This is where we should look at req.user and see if we
+        // can consolidate profiles. We might want to give the developer a hook
+        // so that they can control the consolidation strategy.
+        profile._json.accessToken = accessToken;
+
+        let data = Object.assign({
+          [`${options.provider}Id`]: profile.id,
+          [`${options.provider}`]: profile._json
+        });
+        
+        return app.service(options.userEndpoint).create(data, { internal: true }).then(user => {
+          return done(null, user);
+        }).catch(done);
+      }).catch(done);
```

Comment:
From 60 to here it can probably just be changed to 

```js
return app.service(options.userEndpoint).create(data, { internal: true })).then(user => done(user)).catch(done);
```